packages/bierzo-wallet/src/logic/account.ts

```diff
@@ -0,0 +1,30 @@
+import { Address, ChainId } from '@iov/bcp';
+import { BnsConnection } from '@iov/bns';
+
+import { getConfig } from '../config';
+import { getConnectionFor, isBnsSpec } from './connection';
+
+// getAddressByName returns the address associated with the name, or undefined if not registered
+// the name should not have the "*iov" suffix
+export async function getAddressByName(username: string, chainId: ChainId): Promise<Address | undefined> {
```

Comment:
The existing usernames info are already in the redux store, I think this code should be a selector because we are duplicating effort. Also having centralized the info in one place, making redux the single point of truth for views make the app stronger.

Something like:
```
const getAddressByNameSelector = createSelector(
  state => state.usernames,
  (_, chainId) => chainId,
  (usernames, chainId) => {
       if (!usernames) {
          throw new Error(...);
       }

       const username = usernames.addresses.find(name => name.chainId === chainId)

       if (!username) {
          return undefined;
       }

       return username.address;
}):

....

// In a React component
const address = useSelector(state => getAddressByNameSelector(state, chainId));
```

Reply:
> The existing usernames info are already in the redux store

Those are different usernames. In the redux store we have the names owned by the current user (i.e. the *sender*). But `getAddressByName` makes a lookup in the BNS for the *recipient*.

Reply:
Ahh! Good point. I understand, and in my PR probably I will need to use this method as well.

In this case, the name of the function mislead me, @albertandrejev can you just ignore my comment and rename the function to `lookupRecipientAddressByName`.

Reply:
Good idea 👆